platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/collections/CompoundAssignmentResult.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.file.collections;
+
+import com.google.common.base.Preconditions;
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.internal.file.FileCollectionInternal;
+import org.gradle.api.internal.file.UnionFileCollection;
+import org.gradle.api.internal.tasks.TaskDependencyFactory;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * A helper class to implement an intermediate result of a compound assignment operation, like "+=".
+ * It is then assigned to the left-hand side operand. When the LHS is a ConfigurableFileCollection-typed property of some Gradle-enhanced object, then the assignment action is invoked.
+ */
+class CompoundAssignmentResult extends UnionFileCollection implements org.gradle.api.internal.groovy.support.CompoundAssignmentResult {
```

Comment:
💭 I generally don't like to see same-named classes in gradle/gradle codebase. It trips up quick searches, when only the class name appears in search results, file names and such.

Is there anything preventing us to give this a more explicit name like `FileCollectionCompoundAssignmentResult`?